arch/sim/src/sim/up_interruptcontext.c

```diff
@@ -43,10 +43,47 @@
 #include <nuttx/arch.h>
 #include "up_internal.h"
 
+#ifdef CONFIG_SIM_PREEMPTIBLE
+uint32_t host_signal_save(void);
```

Comment:
@sebastianene07 please merge all preemptive scheduling related change in one patch. Without this patch, your previous change just break the system.

Reply:
Ok I'll squash all these commits in a single one. Is that ok ?

Reply:
Maybe like this:
1.The alarm timer
2.ucontext
3.nsh3 config
But I suggest that:
1.Move alarm timer into up_oneshot.c We don't need create another timer implementation, it's better to enhance up_oneshot.c directly which is much more simple. I suppose you just need add 100~ lines to support this feature.
2.We don't need save/restore context by ucontext API, so it's better to drop ucontext patch.

Reply:
I removed the ucontext API